packages/core/src/components/section/section.tsx

```diff
@@ -190,17 +189,29 @@ export const Section: React.FC<SectionProps> = React.forwardRef((props, ref) =>
                         </>
                     )}
 
-                    {isHeaderRightContainerVisible && (
-                        <div className={Classes.SECTION_HEADER_RIGHT}>
-                            {rightElement}
-                            {collapsible &&
-                                (isCollapsed ? (
-                                    <ChevronDown className={Classes.TEXT_MUTED} />
-                                ) : (
-                                    <ChevronUp className={Classes.TEXT_MUTED} />
-                                ))}
-                        </div>
-                    )}
+                    {collapsible &&
+                        (isCollapsed ? (
+                            <ChevronDown className={Classes.TEXT_MUTED} />
+                        ) : (
+                            <ChevronUp className={Classes.TEXT_MUTED} />
+                        ))}
+                </div>
+            )}
+
+            {rightElement && (
+                // Needs to be outside of the section header because section header may be a button (when collapsable), and
+                // interactive elements (that may be rightElement) can't be inside other buttons.
```

Comment:
> interactive elements (that may be rightElement) can't be inside other buttons

do you have a documentation reference for this?

Reply:
Still waiting on an answer to this question, and my overall question in my last review @bvandercar-vt

Reply:
@bvandercar-vt do you have a good reference for this? a quick search turned up https://accessibilityinsights.io/info-examples/web/nested-interactive/ but not sure how reliable that is/I have not tested myself

Reply:
I'm also not sure this applies here because the header is just a div with an `onClick` event and a `button` role.

I opened a related issue yesterday: https://github.com/palantir/blueprint/issues/6872